interpreter/binary/encode.ml

```diff
@@ -86,6 +86,13 @@ let encode m =
       patch s (p + 3) (lsb ((n lsr 21) lor 0x80));
       patch s (p + 4) (lsb (n lsr 28))
 
+    let bits i j n flags =
```

Comment:
Do you plan to use `bits` and `bit` in other places? If not, please remove them. It takes significant mental effort to understand them. Parameters `i`, `n`, and `flags` kind of make sense, `i` is the insert position, `n` is the source, and `flags` is the definition. `j`, however, does not make sense. It's the number of bits from the left that you expect to be unset.

In essence, it would be helpful if the function signature was self-documenting.